### PR TITLE
Fix hang when returning from shell command

### LINKED ISSUE #3903: Process suspended after shell command execution
**Describe the bug**

When shell command executed with `:` hotkey lazygit process become suspended.

**To Reproduce**

1. Open lazygit
2. Type `:` to enter shell command menu
3. Type any command, ex.: `date`
4. Command output displayed and lazygit process suspended, your shell is open.

**Expected behavior**

Lazygit ask to press any key and after return to lazygit.

**Screenshots**

[demo.webm](https://github.com/user-attachments/assets/42fc45ca-b509-4dac-814e-48449a49bf00)

**Version info:**
_Run `lazygit --version` and paste the result here_

commit=2d0c7cb0fc85e3c262837eb9691813fa97e00e80, build date=2024-09-07T02:26:26Z, build source=binaryRelease, version=0.44.0, os=linux, arch=amd64, git version=2.39.2

_Run `git --version` and paste the result here_

git version 2.39.2

**Additional context**

Problem appears with version 0.44.0 and master.

### PR DESCRIPTION
**Pull request by stefanhaller**
- **PR Description**

In #3793 we changed the execution of shell commands to use an interactive shell (-i), because this allows users to use aliases or shell functions, which is a nice convenience.

Since then, however, many users have reported problems with lazygit not coming back to the foreground after executing a shell command. Some users report that appending "; exit" to the end of the command line solves this. I don't really understand what the cause of this problem was, or why appending "; exit" solves it, but if it helps, let's do it.

Fixes #3903, #3923, #3937.

- **Please check if the PR fulfills these requirements**

* [x] Cheatsheets are up-to-date (run `go generate ./...`)
* [x] Code has been formatted (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#code-formatting))
* [ ] Tests have been added/updated (see [here](https://github.com/jesseduffield/lazygit/blob/master/pkg/integration/README.md) for the integration test guide)
* [ ] Text is internationalised (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#internationalisation))
* [ ] If a new UserConfig entry was added, make sure it can be hot-reloaded (see [here](https://github.com/jesseduffield/lazygit/blob/master/docs/dev/Codebase_Guide.md#using-userconfig))
* [ ] Docs have been updated if necessary
* [x] You've read through your own file changes for silly mistakes etc

### CODE REVIEW

**Comment by rayluo** [comment]
> Some users report that appending "; exit" to the end of the command line solves this. I don't really understand what the cause of this problem was, or why appending "; exit" solves it, but if it helps, let's do it.

I am one of those users commented in #3937. I don't know why or how that `; exit` work, either. I tested your branch (in a github codespace, for what it's worth), but the new change does not seem to work.

```
@rayluo ➜ /workspaces/lazygit (fix-hang-on-shell-command) $ ./lazygit 

+ /bin/bash -i -c ls; exit $?

cmd  CODE-OF-CONDUCT.md  CONTRIBUTING.md  demo  Dockerfile  docs  go.mod  go.sum  lazygit  LICENSE  main.go  Makefile  pkg  README.md  schema  scripts  test  vendor

Press enter to return to lazygit
[1]+  Stopped                 ./lazygit
@rayluo ➜ /workspaces/lazygit (fix-hang-on-shell-command) $ fg
./lazygit


+ /bin/bash -i -c ls; exit $?

cmd  CODE-OF-CONDUCT.md  CONTRIBUTING.md  demo  Dockerfile  docs  go.mod  go.sum  lazygit  LICENSE  main.go  Makefile  pkg  README.md  schema  scripts  test  vendor

Press enter to return to lazygit
[1]+  Stopped                 ./lazygit
@rayluo ➜ /workspaces/lazygit (fix-hang-on-shell-command) $ fg
./lazygit
```

If I manually add yet another "; exit" suffix in my command line, it won't hang.

```
+ /bin/bash -i -c ls -l; exit; exit $?

......
exit

Press enter to return to lazygit
```

**Comment by yam-liu** [comment]
@rayluo It seems the difference is `$?`. Can you remove it from your codespace and test it again?

**Comment by rayluo** [comment]
> @rayluo It seems the difference is `$?`. Can you remove it from your codespace and test it again?

I don't know why `$?` is needed but it turns out it is irrelevant.

The solution is to quote the entire command line. @stefanhaller , please review (and test?) #4129 and merge it to your PR.

**Review by jesseduffield — APPROVED** [review]
LGTM pending @rayluo 's approval

**Comment by rayluo** [comment]
> ... the exit needs to be included in the call to quotedCommandString. Thanks for pointing this out, I included the fix here, so you can close your PR.

OK, that would also mean I will need to wait for next time to get my name shown up in the Lazygit contributors list, but whatever. :-)

Just don't forget to squash your current commit, because its "fixup" keyword probably causes a CI/CD test failure.

> 
> The reason why we want to include the `$?` is so that if a command fails, you 1) get an error message in lazygit, and 2) if you have `promptToReturnFromSubprocess` set to false, it still prompts if the command fails, so that you can read the error output in the console.

Good to know. Thanks for your attention to details!

**Review by rayluo — APPROVED** [review]

**Comment by stefanhaller** [comment]
> Just don't forget to squash your current commit, because its "fixup" keyword probably causes a CI/CD test failure.

Don't worry, I use fixup commits all the time, they are an integral part of my daily workflow, so I know quite well what I have to do with them. 😄

**Pull request merged**
